src/v/cloud_storage/cache_service.h

```diff
@@ -95,6 +96,8 @@ class cache {
     /// only tmp files that are left from previous Red Panda run
     ss::future<> clean_up_at_start();
 
+    ss::future<> recursive_delete_empty_directory(const std::string_view& key);
```

Comment:
I think this method deserves a comment (especially because recursion is "upwards" here, as opposed to downwards, which is more familiar when dealing with filesystem trees).

Reply:
Added a comment